codegen/example/example_server.go

```diff
@@ -278,7 +290,7 @@ func main() {
 			} else if u.Port() == "" {
 				u.Host += ":{{ $u.Port }}"
 			}
-			handle{{ toUpper $u.Transport.Name }}Server(ctx, u, {{ range $.Services }}{{ if .Methods }}{{ .VarName }}Endpoints, {{ end }}{{ end }}&wg, errc, logger, *dbgF)
+			handle{{ toUpper $u.Transport.Name }}Server(ctx, u, tlsCertPath, tlsKeyPath, {{ range $.Services }}{{ if .Methods }}{{ .VarName }}Endpoints, {{ end }}{{ end }}&wg, errc, logger, *dbgF)
```

Comment:
Continuing from my response above, I'm forced to define the TLS paths up front for flag parsing to work correctly. In the case that TLS is not being used, these arguments pass through harmlessly as empty strings, and are further ignored inside `handleHTTPServer`.

If you feel strongly about it, I can define a `handleHTTPServer` with a footprint omitting the arguments, but the implementation of that function would most logically be to simply call `handleHTTPServer` passing empty strings for the TLS paths. I personally did not see much value in that, but I'll make the change if you'd prefer it.

Reply:
As per the above what I'm hoping to avoid is having the code define these flags when no URI uses TLS. Agree that it's really more of a nicety but it doesn't seem too complicated to add and will probably match expectations better.